Dockerfile

```diff
@@ -1,19 +1,21 @@
 FROM node:8-alpine
 
-ARG VERSION
-ARG TYPE=basic
+LABEL maintainer Vincenzo Chianese, vincenzo@express-gateway.io
 
+ENV EG_VERSION 1.7.2
 ENV NODE_ENV production
-# this will enable polling, hot-reload will work on docker or network volumes
+ENV NODE_PATH /usr/local/share/.config/yarn/global/node_modules/
+ENV EG_CONFIG_DIR /var/lib/eg
+# Enable chokidar polling so hot-reload mechanism can work on docker or network volumes
 ENV CHOKIDAR_USEPOLLING true
 
-RUN mkdir -p /usr/src/app
-WORKDIR /usr/src/app
+VOLUME /var/lib/eg
 
-RUN npm install express-gateway@$VERSION && \
-    ./node_modules/.bin/eg gateway create -n gateway -d . -t $TYPE && \
-    npm cache clean --force
+RUN yarn global add express-gateway@$EG_VERSION
```

Comment:
Why use `yarn` here instead of `npm`?

Reply:
I remember there was a reason but I forgot it and I didn't take a note. I'll retry and let you know.

I am quite sure npm was failing because of the way it store the global packages — and it's not `docker-ish`. yarn specifically solved this issue.

Reply:
why use global at all?

Reply:
It was local before, but Kevin was against that because of the weird work dir.
See [here](https://github.com/ExpressGateway/express-gateway/pull/628#discussion_r166337649) for the discussion.

Reply:
To make docker page look normal it has to be another repo, right? so why not commit some generated code into the separate repo. so no need for global or magic path

Reply:
It's not mandatory — but if it makes the thing easier I'm absolutely in for that. What do you have in mind precisely?

Reply:
A new repo containing Docker file, Readme + whatever `eg gateway create` generates 
the last part is refreshed on each version build 

And It will make it easier for customers to build their own images, fork and build your own image

Reply:
So I made a quick test — and unfortunately this approach will have another problem.

The generated `server.js` file is explicitly setting the dir path for the configuration files and models — and will ignore the environment variable (if provided). This means that if I mount my configuration files somewhere and I set the `EG_CONFIG_DIR` variable to the correct directly — the gateway will not honor the request.

Unless we fix that in the code (we're having the discussion in #632) we cannot do that.

Guys — I'd say we can iterate on this with the passing of the time. I'd rush just a bit more to make sure it gets on Docker official registry first. As the checks on their side will require time, we need to start the process as soon as possible.

Reply:
@XVincentX If you think this is good enough to get into the official registry, go for it.  If we make changes, do they also have to go through a complete review process?  If so, I'd say let's wait to get the general approach right.  I think the idea of having a different directory or repo for the Docker image is a good one.  These limitations are superficial and can be overcome easily.

Reply:
I think it's good enough.

I need to resurrect this PR a bit (it was a bit stale) and try to automate couple of things, but I can get this over the finish line tomorrow and merge it.

I've followed all the guidelines so — if nothing goes wrong — it should get accepted. I can't guarantee that, though.